bin/check-strongswan-connection.rb

```diff
@@ -27,23 +27,44 @@
 #
 
 require 'sensu-plugin/check/cli'
+require 'open3'
+require 'set'
 
 # check strongswan connection status
 class CheckStrongswanConnection < Sensu::Plugin::Check::CLI
   option :connection,
          description: 'Connection name to check',
          short: '-c NAME',
-         long: '--connection NAME',
-         required: true
+         long: '--connection NAME'
+
+  def run_ipsec_statusall
+    stdout, result = Open3.capture2("ipsec statusall #{config[:connection]}")
+    unknown 'Unable to get Strongswan statusall' unless result.success?
```

Comment:
I think we should return the result in the error message so it could be easier to troubleshoot issues.

Reply:
fixed
```
    stdout, stderr, result = Open3.capture3("ipsec statusall #{config[:connection]}")
    unknown stderr unless result.success?
```

Reply:
After trying this it looks like we should do a rescue:
```
$ irb
irb(main):001:0> stdout, stderr, result = Open3.capture3("ipsec statusall foo")
NameError: uninitialized constant Open3
	from (irb):1
	from /home/babrams/.rbenv/versions/2.3.5/bin/irb:11:in `<main>'
irb(main):002:0> require 'open3'
=> true
irb(main):003:0> stdout, stderr, result = Open3.capture3("ipsec statusall foo")
Errno::ENOENT: No such file or directory - ipsec
	from /home/babrams/.rbenv/versions/2.3.5/lib/ruby/2.3.0/open3.rb:199:in `spawn'
	from /home/babrams/.rbenv/versions/2.3.5/lib/ruby/2.3.0/open3.rb:199:in `popen_run'
	from /home/babrams/.rbenv/versions/2.3.5/lib/ruby/2.3.0/open3.rb:95:in `popen3'
	from /home/babrams/.rbenv/versions/2.3.5/lib/ruby/2.3.0/open3.rb:258:in `capture3'
	from (irb):3
	from /home/babrams/.rbenv/versions/2.3.5/bin/irb:11:in `<main>'

```

Reply:
I guess other than rescuing the file not found this should work:
```
$ irb
irb(main):001:0> require 'open3'
=> true
irb(main):002:0> stdout, stderr, result = Open3.capture3("ipsec statusall foo")
=> ["", "connecting to 'unix:///var/run/charon.ctl' failed: Permission denied\nfailed to connect to stroke socket 'unix:///var/run/charon.ctl'\n", #<Process::Status: pid 11853 exit 1>]
```

Up to you if you want to rescue that. I will wait and if you don't want to add it just let me know and I can merge as is.

Reply:
Will be honest I am a bit tired and in need of some sleep. I will check on this later today when its not so early for my timezone.